workflow-templates/im-build-db-ci.yml

```diff
@@ -0,0 +1,186 @@
+  pull_request:
+    branches: [ master ]
+env:
+  DB_NAME: '' # TODO: Fill in your repo's database name
+  DB_SERVER_NAME: 'localhost,1433'
+  DEFAULT_BRANCH: 'main'
+
+  # The following are all for the snapshot step below and can be removed if it's not used
+  SNAPSHOT_PATH: './snapshot' # TODO: Verify this path is correct
+  EXCLUDED_DB_SNAPSHOT_OBJECTS: '' # TODO: Add some comma separated values to this
+
+  # These next variables are for backing up the database. Remove if not needed.
+  BACKUP_PATH: '' # TODO: Update or remove this
+  BACKUP_FILENAME: '' # TODO: Update or remove this
+
+jobs:
+  # TODO: Remove this and it's usages below if no snapshots are created for your database
+  check-for-snapshot-commit:
+    runs-on: ubuntu-latest
+    outputs:
+      should-skip-remaining-workflow-jobs: ${{ steps.check.outputs.skip }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check for snapshot commit
+        id: check
+        shell: pwsh
+        run: |
+          $CurrentBranch = git rev-parse --abbrev-ref HEAD
+          $LastCommitBy = git --no-pager show -s --format="%aN"
+
+          if ($LastCommitBy -eq "github-actions" -and $CurrentBranch -ne "${{ env.DEFAULT_BRANCH }}")
+          {
+            Write-Host "Snapshot was last commit, skipping the other jobs"
+            echo "::set-output name=skip::true"
+          }
+          else
+          {
+            Write-Host "Snapshot was not the last commit, continuing with the other jobs"
+            echo "::set-output name=skip::false"
+          }
+
+  # TODO: Remove this linting job if your project doesn't need/want it
+  lint-migration-files:
+    runs-on: ubuntu-latest
+    needs: check-for-snapshot-commit
+    if: ${{ needs.check-for-snapshot-commit.outputs.should-skip-remaining-workflow-jobs != 'true' }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - id: migration-folder
+        shell: pwsh
+        run: echo "::set-output name=folder::$($(Get-Date).Year).$($(Get-Date).Month.ToString("00"))"
+      - name: SQL Lint
+        uses: im-open/tsql-lint-action@v1.0.0
+        with:
+          tsqllint-version: 1.11.0
+          path-to-sql-files: "Database/src/SqlDatabase/Migrations/${{ steps.migration-folder.outputs.folder }}" # TODO: Update this path to wherever the files to lint live
+          file-name-filter: "V9*.sql" # TODO: Update this if necessary
+          path-to-lint-config: ./.tsqllintrc # TODO: Update this with the path to your project's .tsllintrc file
+
+
+  build-database:
+    runs-on: [self-hosted, windows-2019-dev]
+    needs: [check-for-snapshot-commit, lint-migration-files]
+    if: ${{ needs.check-for-snapshot-commit.outputs.should-skip-remaining-workflow-jobs != 'true' }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Flyway
+        uses: im-open/setup-flyway@v1.0.0
+        with:
+          version: 7.2.0
```

Comment:
Do we want a TODO here for the version number?

Reply:
This is the only one I know for sure works. With all of our windows vm problems I could only try a few others out, so there may be others, but I don't know if we want people changing this without knowing that they might run into problems. It's probably worth adding a comment saying something along those lines...